buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -114,6 +116,17 @@ class BuildPlugin implements Plugin<Project> {
                 throw new GradleException("Java ${minimumJava} or above is required to build Elasticsearch")
             }
 
+            // TODO: remove when the build requires Java 9
```

Comment:
instead of the TODO, we can add a failure based on `minimumJava.isJava8Compatible()`. Then, when we change `minimumJava`, we get the reminder for free ;-)